p2p/host/basic/basic_host.go

```diff
@@ -174,6 +175,8 @@ func (h *BasicHost) NewStream(pid protocol.ID, p peer.ID) (inet.Stream, error) {
 // h.Network.Dial, and block until a connection is open, or an error is
 // returned. // TODO: Relay + NAT.
 func (h *BasicHost) Connect(ctx context.Context, pi peer.PeerInfo) error {
+	ctx, cancel := context.WithTimeout(ctx, host.DiscoveryConnTimeout)
+	defer cancel()
```

Comment:
So far in the code the timeouts either have the same value, or none at all.
Rather than having to reason the timeout each time, why not localize to the place where the ctx and timeout are needed and specify the timeout here?